src/base/static/libs/maps/index.js

```diff
@@ -357,14 +337,11 @@ class MainMap {
       this.layers[config.id] = this.map.createWMTSLayer(config);
       this.map.addLayer(this.layers[config.id]);
     } else if (config.type === "vector-tile") {
-      const vectorLayerGroupConfig = await this.map.createVectorTileLayer(
-        config,
-      );
-      this.layers[config.id] = vectorLayerGroupConfig;
-      this.map.addVectorLayerGroup(this.layers[config.id]);
+      this.layers[config.id] = await this.map.createVectorTileLayer(config);
+      this.map.addLayer(this.layers[config.id]);
     } else if (config.type === "json") {
       this.layers[config.id] = this.map.createGeoJSONLayer(config);
```

Comment:
> we are planning to store the Map Provider layer representation in our store?

No-- my understanding is that the state in the store will either be the raw layer config, or perhaps some minimalistic object that keeps track of layer id, visibility, etc.

For now the generated layer objects live in `this.layers` in the `MainMap` class, but if I'm understanding correctly it would be better for them to be cached in the Provider? Maybe this cache could have the following structure:
```
{
  "<layer-id-from-config-a>": [
    <internal-layer-id-1>, 
    <internal-layer-id-2>,
    etc.
  ],
  "<layer-id-from-config-b>": [...],
}
```

So what this buys us is that the internal Mapbox layers are encapsulated in the Provider, and our Redux state knows only about the config-supplied layer id. The map layer duck would use these ids to manipulate groups of Mapbox layers to give the illusion that a single config layer is a single Provider layer.

Reply:
EDIT: actually I guess we'd need another structure to cache the internal layers themselves, but this could essentially be the same thing as `this.layers`, just encapsulated in the provider.

Reply:
Yes, "generated layer objects" === map provider representations. And "Provider" === "MapProvider" in the comments above.  Sorry, I should use one piece of terminology :/

> But I still don't even see why we would ever need to access the map provider representation of the layer outside of the Map Provider

I don't think we will. Only the provider will know about the internal representations of layers. We'll need to tie our first-class representations of layers to the internal Map Provider-specific components, but that's what I imagine the bookkeeping object above will accomplish for us.

But just so I'm crystal clear, what we're currently storing in `this.layers` is **not** our first class representation-- it's the map provider representation. The way we've been discussing it I think `this.layers` would be a copy of the `layerConfig` if it is indeed our store for first-class layer representations.

Or is that off base?

Reply:
>We'll need to tie our first-class representations of layers to the internal Map Provider-specific layers

If we really need to tie these together, we can store a "config id to representation id" mapping as you described above. But it should live within the Map Provider.

It seems like the only reason we need this is to support accessing the multiple symbol/line/fill map provider layers, as you just mentioned. I think there is a better abstraction here, but let's punt that for now.

>But just so I'm crystal clear, what we're currently storing in this.layers is not our first class representation-- it's the map provider representation. The way we've been discussing it I think this.layers would be a copy of the layerConfig if it is indeed our store for first-class layer representations.

Yes, that's right.